cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppFabricHttpHandler.java

```diff
@@ -1230,8 +1230,12 @@ private void programListByDataAccess(HttpRequest request, HttpResponder responde
     }
   }
 
+  /**
+   * @return list of program records; or null if the stream or dataset does not exist.
```

Comment:
though do you need to indicate that the list is empty if no programs were found yet the data exists?